modules/dnn/src/layers/fast_convolution/depthwise_convolution.cpp

```diff
@@ -84,19 +84,20 @@ static void depthWiseBlockConv2D(const float* wptr,
         v_float32x4 vbias = v_setall_f32(bias);
         v_float32x4 vrc = v_setall_f32(relu_coeff);
 
-        if (stride_w == 1 || (stride_w == 2 && dilation_w == 1))
+        if ((stride_w == 1 || (stride_w == 2 && dilation_w == 1)) && outW1 >= VEC_NLANES)
         {
-            if( stride_w == 1 )
+            if (stride_w == 1)
             {
-                for( ; out_j < outW1; out_j += VEC_NLANES )
+                for (; out_j < outW1; out_j += VEC_NLANES)
                 {
-                    if (out_j + VEC_NLANES > outW1)
-                    {
-                        if (out_j <= pad_l || outW1 - VEC_NLANES < 0)
-                            break;
+                    if (out_j > outW1 - VEC_NLANES)
                         out_j = outW1 - VEC_NLANES;
-                    }
+
                     int in_j = out_j * stride_w - pad_l;
+
+                    if (in_j < 0)
+                        break;
```

Comment:
That SIMD bailout code looks strange.
There is no guarantee that `out_j` value is valid and consistent after that. But that value is used in scalar code below (after `#endif`), so this is critical:

```
for (; out_j < outW1; out_j++)
```

**Requirement**: "Scalar" code should NOT recompute the same values again after SIMD code.

---

What values of input parameters trigger that case? (here and below with stride=2)
(currently this bailout is not triggered by tests)

Also there is no similar check in "scalar" part.

Reply:
Hi @alalek thanks for your suggestion, I tried simplifying the code further.